src/main.rs

```diff
@@ -48,14 +55,23 @@ impl fmt::Display for Repo {
 }
 
 impl Repo {
-    fn new(name: &str) -> Self {
-        let name = name.to_string();
+    fn new(name: impl Into<String>) -> Self {
         Repo {
-            name,
+            name: name.into(),
             total_commits: 0,
             total_curses: 0,
+            authors: HashMap::new(),
         }
     }
+
+    fn author_for(&mut self, author_name: &str) -> &mut Author {
+        if !self.authors.contains_key(author_name) {
+            self.authors
+                .insert(author_name.to_owned(), Author::new(author_name));
+        }
+
+        self.authors.get_mut(author_name).expect("exists")
+    }
```

Comment:
This can be shortened to the below:
`self.authors.entry(author_name.into()).or_insert(Author::new(author_name))`

This makes it a more idiomatic method for a "get or create" pattern.

Reply:
I considered the approach, but I really wanted to not allocated a new `String` unless it was strictly necessary (it's only needed when inserting into the HashMap). This is a long standing problem, I would love to know if there's an easier way to do this with stdlib or if there's a crate that solves this (see: https://users.rust-lang.org/t/efficient-string-hashmaps-for-a-frequency-count/7752/8).

Additionally, you would probably want to use `or_insert_with` instead of `or_insert` as the `Author::new(..)` is evaluated eagerly for each call with `or_insert` (and `Author::new` copies a string).

Something like: `self.authors.entry(author_name.into()).or_insert_with(|| Author::new(author_name))`.